File: source/RevitDevTool.Test/XYZVisualization.cs
using System.Diagnostics;
using Autodesk.Revit.Attributes;
using Autodesk.Revit.UI.Selection;
using Nice3point.Revit.Toolkit.External;

namespace RevitDevTool.Test;

[Transaction(TransactionMode.Manual)]
[UsedImplicitly]
public class XyzVisualization : ExternalCommand
{
    public override void Execute()
    {
        try
        {
            var xyz = UiDocument.Selection.PickObject(ObjectType.PointOnElement);
            var xyzPoint = xyz.GlobalPoint;
            Trace.Write(xyzPoint);
        }
        catch (Exception e)
        {
            Trace.TraceError($"Error in XyzVisualization: {e.Message}");
        }
    }
}

[Transaction(TransactionMode.Manual)]
[UsedImplicitly]
public class XyzsVisualization : ExternalCommand
{
    public override void Execute()
    {
        try
        {
            var xyzRefs = UiDocument.Selection.PickObjects(ObjectType.PointOnElement);
            var xyzs = xyzRefs.Select(x => x.GlobalPoint).ToList();
            if (xyzs.Count == 0)
            {
                Trace.TraceWarning("No points selected.");
                return;
            }
            Trace.Write(xyzs);
        }
        catch (Exception e)
        {
            Trace.TraceError($"Error in XyzsVisualization: {e.Message}");
        }
    }
}

File: source/Samples/FSharpDemo/XYZVisualization.fs
namespace FSharpDemo
open System.Diagnostics
open System.Linq
open Autodesk.Revit.Attributes
open Autodesk.Revit.UI.Selection
open JetBrains.Annotations
open Nice3point.Revit.Toolkit.External

[<Transaction(TransactionMode.Manual)>]
[<UsedImplicitly>]
type XyzVisualization() =
    inherit ExternalCommand()

    override this.Execute() =
        try
            let xyz = this.UiDocument.Selection.PickObject(ObjectType.PointOnElement)
            Trace.Write(xyz.GlobalPoint)
        with
        | ex -> Trace.TraceError($"Error in XyzVisualization: {ex.Message}")

[<Transaction(TransactionMode.Manual)>]
[<UsedImplicitly>]
type XyzsVisualization() =
    inherit ExternalCommand()

    override this.Execute() =
        try
            let xyzRefs = this.UiDocument.Selection.PickObjects(ObjectType.PointOnElement)
            let xyzs = xyzRefs |> Seq.map (fun x -> x.GlobalPoint) |> Seq.toList
            if xyzs.IsEmpty then
                Trace.TraceWarning("No points selected.")
            else
                Trace.Write(xyzs)
        with
        | ex -> Trace.TraceError($"Error in XyzsVisualization: {ex.Message}")

